Packs/mnemonicMDR/Integrations/ArgusManagedDefence/ArgusManagedDefence.py

```diff
@@ -204,46 +255,270 @@ def test_module_command() -> str:
 
 
 def fetch_incidents(
-    last_run: dict, first_fetch_period: str, limit: int = 25, min_severity: str = "low"
+    last_run: dict,
+    first_fetch_period: str,
+    limit: int = 25,
+    min_severity: str = "low",
+    integration_instance: str = "",
+    mirror_direction: str = "None",
+    mirror_tags: str = "argus_mirror",
+    exclude_tag: str = "",
 ):
     start_timestamp = last_run.get("start_time", None) if last_run else None
+    # Exclude closed cases
+    sub_criteria = [{"exclude": True, "status": ["closed"]}]
+    # Exclude cases with {key} or {key: value} tags
+    if exclude_tag:
+        tag_list = exclude_tag.strip().split(",")
+        if len(tag_list) == 1:
+            sub_criteria.append({"exclude": True, "tag": {"key": tag_list[0]}})
+        elif len(tag_list) == 2:
+            sub_criteria.append(
+                {"exclude": True, "tag": {"key": tag_list[0], "values": tag_list[1]}}
+            )
     # noinspection PyTypeChecker
     result = advanced_case_search(
         startTimestamp=start_timestamp if start_timestamp else first_fetch_period,
         endTimestamp="now",
         limit=limit,
         sortBy=["createdTimestamp"],
         priority=build_argus_priority_from_min_severity(min_severity),
-        subCriteria=[
-            {"exclude": True, "status": ["closed"]},
-        ],
+        subCriteria=sub_criteria,
         timeFieldStrategy=["createdTimestamp"],
     )
     incidents = []
-    for case in result["data"]:
-        incidents.append(
+    for case in result.get("data", []):
+        case["xsoar_mirroring"] = {
+            "dbotMirrorId": str(case["id"]),
+            "dbotMirrorInstance": integration_instance,
+            "dbotMirrorDirection": MIRROR_DIRECTION[mirror_direction],
+            "dbotMirrorTags": argToList(mirror_tags),
+        }
+        case["url"] = f"https://portal.mnemonic.no/spa/case/view/{case['id']}"
+        incident = {
+            "name": f"#{case['id']}: {case['subject']}",
+            "occurred": case["createdTime"],
+            "severity": argus_priority_to_demisto_severity(case["priority"]),
+            "status": argus_status_to_demisto_status(case["status"]),
+            "details": json.dumps(case),
+            "rawJSON": json.dumps(case),
+        }
+        incidents.append(incident)
+
+    if result.get("data", []):
+        last_run["start_time"] = str(result.get("data")[-1]["createdTimestamp"] + 1)
```

Comment:
Are you sure that this won't cause us to miss any incidents?

Reply:
Yes. Otherwise I will get duplicates.

Reply:
But wouldn't it cause you to miss incidents? What happens if there's an incident that needs to be fetched that its start time is just between the last `start_time` and the second you added?

Reply:
Yes. We use epoch milliseconds. This logic has been tested. If we don't use `+1` we will get duplicates.